src/services/study/network-map.js

```diff
@@ -137,3 +146,93 @@ export function fetchAllCountries(studyUuid, currentNodeUuid) {
     console.debug(fetchCountriesUrl);
     return backendFetchJson(fetchCountriesUrl);
 }
+
+function createEquipmentIdentifierList(equipmentType, equipmentList) {
+    return {
+        type: 'IDENTIFIER_LIST',
+        equipmentType: equipmentType,
+        filterEquipmentsAttributes: equipmentList.map((eqId) => {
+            if (eqId?.id) {
+                return { equipmentID: eqId.id };
+            } else {
+                return { equipmentID: eqId };
+            }
+        }),
+    };
+}
+export async function createMapFilter(
+    filter,
+    distDir,
+    studyUuid,
+    currentNodeUuid,
+    selectedEquipements
+) {
+    let equipementList = [];
+    switch (filter.equipmentType) {
+        case EQUIPMENT_TYPES.SUBSTATION:
+        case EQUIPMENT_TYPES.LINE:
+            equipementList = createEquipmentIdentifierList(
+                filter.equipmentType,
+                selectedEquipements
+            );
+            break;
+
+        case EQUIPMENT_TYPES.VOLTAGE_LEVEL:
+        case EQUIPMENT_TYPES.BUS:
+        case EQUIPMENT_TYPES.TWO_WINDINGS_TRANSFORMER:
+        case EQUIPMENT_TYPES.THREE_WINDINGS_TRANSFORMER:
+        case EQUIPMENT_TYPES.BUSBAR_SECTION:
+        case EQUIPMENT_TYPES.GENERATOR:
+        case EQUIPMENT_TYPES.BATTERY:
+        case EQUIPMENT_TYPES.LOAD:
+        case EQUIPMENT_TYPES.SHUNT_COMPENSATOR:
+        case EQUIPMENT_TYPES.DANGLING_LINE:
+        case EQUIPMENT_TYPES.STATIC_VAR_COMPENSATOR:
+        case EQUIPMENT_TYPES.HVDC_CONVERTER_STATION:
+        case EQUIPMENT_TYPES.VSC_CONVERTER_STATION:
+        case EQUIPMENT_TYPES.LCC_CONVERTER_STATION:
+        case EQUIPMENT_TYPES.SWITCH:
+            const substationsIds = selectedEquipements.map(
+                (substation) => substation.id
+            );
+            if (substationsIds.length === 0) {
+                throw new Error('No substations selected');
+            }
+            try {
+                const elements = await fetchEquipmentsIds(
+                    studyUuid,
+                    currentNodeUuid,
+                    substationsIds,
+                    filter.equipmentType,
+                    false
+                );
+
+                if (elements.length > 0) {
+                    equipementList = createEquipmentIdentifierList(
+                        filter.equipmentType,
+                        elements
+                    );
+                }
+            } catch (error) {
+                throw error;
+            }
+
+            break;
+
+        default:
+            break;
+    }
+
+    if (equipementList.filterEquipmentsAttributes?.length === 0) {
+        throw new Error('No equipment selected');
+    }
+
+    if (equipementList.filterEquipmentsAttributes.length > 0) {
+        await createFilter(
```

Comment:
Deal with errors ?

Reply:
i deal with the errors in the caller 'map-viwer.jsx'